device/iot-device-samples/handle-messages/src/main/java/samples/com/microsoft/azure/sdk/iot/HandleMessages.java

```diff
@@ -223,7 +223,7 @@ else if (protocolStr.equals("mqtt_ws"))
 
         System.out.println("Opened connection to IoT Hub.");
 
-        System.out.println("Beginning to receive messages...");
+        System.out.println("Beginning to receive messages(only for MQTT and AMQP). To receive in Https, send message and then start the sample...");
```

Comment:
I didn't get this point - does this refer to the sequence of sending message from service and then setting callback on the device?

Reply:
yes. For MQTT and AMQP, you can run the device sample and then send messages from the portal or wherever you want. However for https, it is not really polling. You need to send the message and then execute receive. If you run this sample and send a message and wait for the message to show up, it will not show up.

Reply:
That is interesting!

Reply:
If you think it is a bug, we should discuss.